lib/proxysql_utils.cpp

```diff
@@ -431,3 +432,47 @@ void close_all_non_term_fd(std::vector<int> excludeFDs) {
 		}
 	}
 }
+
+vector<string> sort_versions(vector<string> versions) {
+	std::sort(
+		versions.begin(), versions.end(),
+		[](const string& v1, const string& v2) {
+			const auto result =
+				mismatch_(
+					v1.cbegin(), v1.cend(), v2.cbegin(), v2.cend(),
+					[](const unsigned char lhs, const unsigned char rhs) {
+						return tolower(lhs) == tolower(rhs);
+					}
+				);
+
+			const bool not_equal = result.second != v2.cend();
+			const bool fst_shorter = result.first == v1.cend();
+			const bool fst_lesser = std::tolower(*result.first) < std::tolower(*result.second);
+
+			return not_equal && (fst_shorter || fst_lesser);
+		}
+	);
+
+	return versions;
+}
+
+std::pair<int,const char*> get_dollar_quote_error(const char* version) {
```

Comment:
At present, I would only check for a version number starting with `8.1` .
I have no reason to believe that the same should work for MySQL 8.1.1 , 8.1.2 , etc .
Also, we need to support version numbers like `X.Y.Z-extra` , for example `8.1.0-0ubuntu0.22.04.2` .

That said, at present `sort_versions()` seems not needed

Reply:
I have simplified the check, now it only check that it starts with `8.1`.